packages/graphql/src/schema/resolvers/query/fulltext.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ObjectTypeComposerFieldConfigDefinition } from "graphql-compose";
+import type { GraphQLResolveInfo } from "graphql";
+import { execute } from "../../../utils";
+import { translateRead } from "../../../translate";
+import type { Node } from "../../../classes";
+import type { Context, FulltextIndex } from "../../../types";
+import getNeo4jResolveTree from "../../../utils/get-neo4j-resolve-tree";
+import Cypher from "@neo4j/cypher-builder";
+
+export function fulltextResolver(
+    { node }: { node: Node },
+    index: FulltextIndex
+): ObjectTypeComposerFieldConfigDefinition<any, any, any> {
+    async function resolve(_root: any, args: any, _context: unknown, info: GraphQLResolveInfo) {
+        const context = createFulltextContext(index, args, _context, info);
+        const { cypher, params } = translateRead({ context, node }, node.singular);
+        const executeResult = await execute({
+            cypher,
+            params,
+            defaultAccessMode: "READ",
+            context,
+        });
+        return executeResult.records;
+    }
+
+    return {
+        type: `[${node.fulltextTypeNames.result}!]!`,
+        resolve,
+        args: {
+            phrase: "String!",
+            where: node.fulltextTypeNames.where,
+            sort: `[${node.fulltextTypeNames.sort}!]`,
+            limit: "Int",
+            offset: "Int",
+        },
+    };
+}
+
+function createFulltextContext(index: FulltextIndex, args: any, _context: unknown, info: GraphQLResolveInfo): Context {
+    const context = _context as Context;
+    context.resolveTree = getNeo4jResolveTree(info, { args });
+    context.resolveTree.args.options = {
+        sort: context.resolveTree.args.sort,
+        limit: context.resolveTree.args.limit,
+        offset: context.resolveTree.args.offset,
+    };
```

Comment:
I don't think I understand this comment?

Is it not cleaner to do it here? Literally, every helper function expects these arguments to be in this format so why not do it just once in a function where it's clear that it's just doing this for full-text?

Reply:
Well the resolve tree is kind of "sacred" - it's the representation of the incoming GraphQL query. It feels like we're having to "hack" it because we're using the `translateRead` function. It's not a massive problem, just feels a little bit dirty, we should perhaps look at removing these helper functions reliance on the resolve tree itself in future.